src/create_demos.js

```diff
@@ -1,6 +1,81 @@
+var fbpManifest = require('fbp-manifest');
 var path = require('path');
+var bluebird = require('bluebird');
 
-module.exports = function (grunt, graphs, options, callback) {
+var filterDependencies = function(modules, options, callback) {
+  var applicableModules = modules.filter(function (m) {
+    var modulePath = path.resolve(options.baseDir, m.base);
+    if (modulePath !== options.baseDir) { return false; }
+    if (m.runtime !== 'noflo-browser' && m.runtime !== 'noflo') { return false; }
+    return true;
+  });
+  applicableModules.forEach(function (m) {
+    m.components = m.components.filter(function (c) {
+      if (c.elementary) {
+        return false;
+      }
+      if (path.extname(c.path) !== '.json') {
+        return false;
+      }
+      var moduleName = m.name + '/' + c.name;
+      if (options.graph && moduleName !== options.graph) {
+        return false;
+      }
+      return true;
+    });
+  });
+  applicableModules = applicableModules.filter(function (m) {
+    if (!m.components || !m.components.length) { return false; }
+    return true;
+  });
+  callback(null, applicableModules);
+};
+
+var serialize = function (module, options, grunt, callback) {
+  var modulePath = path.resolve(options.baseDir, module.base);
+  var templateName = (options.debug) ? 'graphDebug' : 'graph';
+  var graphFileTemplatePath = path.resolve(__dirname, '../templates/' + templateName + '.html');
+  var graphFileTemplate = grunt.file.read(graphFileTemplatePath);
+  module.components.forEach(function (graphComponent) {
+    var graphPath = path.resolve(modulePath, graphComponent.path);
+    var graph = grunt.file.readJSON(graphPath);
+    if (!graph.properties || !graph.properties.environment || !graph.properties.environment.content) {
+      // No HTML demo definition
+      return;
+    }
+    var templateData = {
+      name: graphComponent.name,
+      scripts: options.graph_scripts,
+      lib: module.name,
+      ideUrl: options.ide,
+      signalServer: options.signalserver,
+      noflo: options.destName + '.js',
+      graphPath: module.name + '/' + graphComponent.name,
+      content: graph.properties.environment.content,
+      heads: options.heads
+    };
+    var templated = grunt.template.process(graphFileTemplate, {
+      data: templateData
+    });
+    var demoFile = path.resolve(options.destDir, graphComponent.name + '.html');
+    grunt.file.write(demoFile, templated);
+    grunt.log.writeln('Demo file ' + demoFile + ' built');
+  });
+  callback(null);
+};
+
+var discover = function (options, callback) {
+  var manifestOptions = options.manifest;
+  fbpManifest.load.load(options.baseDir, manifestOptions, function(err, manifest) {
+    if (err) {
+      return callback(err);
+    }
+    filterDependencies(manifest.modules, options, callback);
```

Comment:
Dependency handling is done in `src/build_loader.js` where it includes all installed components (except if a graph is specified, in which case only deps of that graph).

Here we filter down to only graphs of the local project to build HTML files for.